src/renderer/atlas/BackendD3D.cpp

```diff
@@ -305,6 +306,37 @@ void BackendD3D::_updateFontDependents(const RenderingPayload& p)
 {
     const auto& font = *p.s->font;
 
+    // The max height of Curly line peak in `em` units.
+    const auto maxCurlyLinePeakHeightEm = 0.075f;
+    // We aim for atleast 1px height, but since we draw 1px smaller curly line,
+    // we aim for 2px height as a result.
+    const auto minCurlyLinePeakHeight = 2.0f;
+
+    // Curlyline uses the gap between cell bottom and singly underline position
+    // as the height of the wave's peak. The baseline for curly-line is at the
+    // middle of singly underline. The gap could be too big, so we also apply
+    // a limit on the peak height.
+    const auto strokeHalfWidth = font.underline.height / 2.0f;
+    const auto underlineMidY = font.underline.position + strokeHalfWidth;
+    const auto cellBottomGap = font.cellSize.y - underlineMidY - strokeHalfWidth;
+    const auto maxCurlyLinePeakHeight = maxCurlyLinePeakHeightEm * font.fontSize;
+    auto curlyLinePeakHeight = std::min(cellBottomGap, maxCurlyLinePeakHeight);
+
+    // When it's too small to be curly, make it straight.

```

Comment:
clever!

Reply:
Hi guys,

Something isn't right here: I can't see curly underlines until I change font size to at least 35 in conhost or 20 in WT, which is huge even on a full HD screen:

Conhost (20) | WT (12.8107)
![image](https://github.com/microsoft/terminal/assets/11453922/4a03c331-1690-4ade-b352-8a7e9353ec43)

It's definitely not too small to be curly yet:
wezterm (8.0)
![image](https://github.com/microsoft/terminal/assets/11453922/419b48b2-42a0-4a78-b4d2-03aa670b0d76)

Reply:
This seems to be the case for me too when using a 1440p monitor.

Reply:
Curly line is only drawn when there's enough space for it [^1] - space between (singly) underline and Cell bottom. If we don't do this, the curly line will always be rendered, but it would be clipped, which wouldn't look nice (or correct) 😅
For example: 
When using *Courier New* (font)

**When WT draws Curlyline**:

![image](https://github.com/microsoft/terminal/assets/55626797/3f64209f-a7d7-42da-8a0c-9a5e010de5fd)

**When it doesn't**:

![Screenshot 2023-11-10 204936](https://github.com/microsoft/terminal/assets/55626797/a0c87a49-86d2-4b0a-8860-1157e5bec7af)

I'd love to help you guys 😀, but can you file an issue for this? We can discuss in detail what might be causing this issue at those font sizes. Also, make sure to add font name (and other details that might be helpful) for repro.

[^1]: To get a rough idea of how much space is available between cell bottom and underlines is to select the cell where the underline is drawn. Cell area is highlighted by the selection, and you can see how much space is available between cell bottom and underline baseline. This works on Conhost too (even better since you can scroll to increase fontsize, and the selection is preserved).

Reply:
> can you file an issue for this?

Sure, #16288